src/lib/renderElement.js

```diff
@@ -3,53 +3,151 @@ import { addEvent, removeEvent, setupEventListeners } from './eventManager';
 import { createElement__v2 } from "./createElement__v2.js";
 
 // TODO: processVNode 함수 구현
-function processVNode() {
+function processVNode(vNode) {
   // vNode를 처리하여 렌더링 가능한 형태로 변환합니다.
   // - null, undefined, boolean 값 처리
   // - 문자열과 숫자를 문자열로 변환
   // - 함수형 컴포넌트 처리 <---- 이게 제일 중요합니다.
   // - 자식 요소들에 대해 재귀적으로 processVNode 호출
+
+
+  if(!vNode || typeof vNode === 'boolean') {
+    return document.createTextNode('')
+  }
+
+  if(typeof vNode === 'string' || typeof vNode === 'number') {
+    return String(vNode)
+  }
+
+  if(typeof vNode.type === 'function') {
+    //어렵다 함수형 컴포넌트..
+    const result = vNode.type(vNode.props || {});
+    return processVNode(result);
+  }
+
+  // 자식요소
+  const proChildren = (vNode.children || []).map((child) => processVNode(child));
+
+  return {
+    type: vNode.type,
+    props: vNode.props || {},
+    children: proChildren,
+  };
+
 }
 
+
 // TODO: updateAttributes 함수 구현
-function updateAttributes() {
+
+function updateAttributes(target, oldProps, newProps) {
   // DOM 요소의 속성을 업데이트합니다.
   // - 이전 props에서 제거된 속성 처리
   // - 새로운 props의 속성 추가 또는 업데이트
   // - 이벤트 리스너, className, style 등 특별한 경우 처리
+
   //   <이벤트 리스너 처리>
   //     - TODO: 'on'으로 시작하는 속성을 이벤트 리스너로 처리
   //     - 주의: 직접 addEventListener를 사용하지 않고, eventManager의 addEvent와 removeEvent 함수를 사용하세요.
   //     - 이는 이벤트 위임을 통해 효율적으로 이벤트를 관리하기 위함입니다.
+
+
+  oldProps = oldProps || {};
+  newProps = newProps || {};
+
+
+  for (const key in oldProps) {
+    if (!(key in newProps)) {
+      if (key === 'className') {
+        target.className = '';
+      } else if (key.startsWith('on')) {
+        const eventName = key.slice(2).toLowerCase();
+
+        removeEvent(target, eventName, oldProps[key]);
+        // 새로운 이벤트 핸들러가 없으면 제거
+
+      } else if (key === 'style') {
+        Object.assign(target.style, {});
+      } else {
+        target.removeAttribute(key);
+      }
+    }
+  }
+
+  for (const [key, value] of Object.entries(newProps)) {
+    if (value !== oldProps[key]) {
+      if (key === 'className') {
+        target.className = value;
+      } else if (key.startsWith('on')) {
+        const eventName = key.slice(2).toLowerCase();
+        addEvent(target, eventName, value);
```

Comment:
이벤트리스너 등록할 때, function 타입인지 확인하는 부분이 있으면 더 좋을 것 같아용 ㅎㅎ

```suggestion
      } else if (key.startsWith('on') && typeof value === 'function') {
        const eventName = key.slice(2).toLowerCase();
        addEvent(target, eventName, value);
```